src/nimlsp.nim

```diff
@@ -57,8 +57,8 @@ template textDocumentRequest(message, kind, name, body) {.dirty.} =
         filestash = storage / (hash(fileuri).toHex & ".nim" )
       debugEcho "Got request for URI: ", fileuri, " copied to " & filestash
       let
-        rawLine = name["position"]["line"].getInt
-        rawChar = name["position"]["character"].getInt
+        rawLine = max(name["position"]["line"].getInt, 0)
+        rawChar = max(name["position"]["character"].getInt, 0)
```

Comment:
This change means that `rawLine` and `rawChar` aren't really `raw` any longer so it should be renamed to something else.